app/views/support/ABTest.scala

```diff
@@ -90,27 +107,29 @@ case class ChosenVariants(v1: AmountHighlightTest.Variant, v2: MessageCopyTest.V
 }
 
 object Test {
-  def pickRandomly(test: TestTrait): test.Variant = {
-    val n = scala.util.Random.nextDouble()
-    test.weightedVariants.dropWhile(_._1 < n).head._2
-  }
 
-  def pickByQueryStringOrCookie[A](request: Request[A], test: TestTrait): Option[test.Variant] = {
-    val search: Option[String] = request.getQueryString(test.slug)
-      .orElse(request.cookies.get(test.slug + "_GIRAFFE_TEST").map(_.value))
-    test.variants.list.find(_.variantSlug == search.getOrElse(None))
-  }
+  def pickVariant[A](countryGroup: CountryGroup, request: Request[A], test: TestTrait): test.Variant = {
+
+    def pickRandomly: test.Variant = {
+      val n = scala.util.Random.nextDouble()
+      test.weightedVariantsFor(countryGroup).dropWhile(_._1 < n).head._2
```

Comment:
This will crash horribly if we forget to include a variant that applies to the current countryGroup. 
If we want to prevent this we could either force the app to crash on startup if we end up with an empty list for any of the supported countryGroups or do headOption instead of head here and fall back to some default value